src/domains/util.cpp

```diff
@@ -38,34 +38,36 @@ void extend_expr_types(exprt &expr)
     return;
   if(expr.id()==ID_unary_minus)
   {
-    extend_expr_types(expr.op0());
+    exprt &op0=to_unary_minus_expr(expr).op();
```

Comment:
`op0` is a weird variable name since we have an unary operator here. Better use `op` instead.

Reply:
Fixed also a bit lower in the method with typecast expression (also unary). Rebased and squashed.